src/main/scala/mesosphere/marathon/metrics/SlidingAverageSnapshot.scala

```diff
@@ -0,0 +1,133 @@
+package mesosphere.marathon
+package metrics
+
+import java.time.Duration
+import java.util.logging.Logger
+
+import akka.Done
+import akka.actor.{ Actor, ActorRefFactory, Props }
+import kamon.Kamon
+import kamon.metric.{ Entity, EntitySnapshot }
+import com.typesafe.config.Config
+import kamon.metric.SubscriptionsDispatcher.TickMetricSnapshot
+import kamon.metric.instrument.CollectionContext
+import kamon.util.{ MapMerge, MilliTimestamp }
+import mesosphere.marathon.metrics.Metrics.subscribe
+
+/**
+  * Calculates sliding average entity snapshot
+  */
+class SlidingAverageSnapshot(val averagingWindow: Duration) {
+  private val logger: Logger = Logger.getLogger(getClass.getName)
+
+  val collectionContext: CollectionContext = Kamon.metrics.buildDefaultCollectionContext
+
+  /**
+    * The current summarised snapshot
+    */
+  private var currentSnapshot: TickMetricSnapshot = TickMetricSnapshot(MilliTimestamp.now, MilliTimestamp.now, Map.empty)
+
+  /**
+    * A ring buffer that collects the snapshots
+    */
+  private var averageRing: Array[TickMetricSnapshot] = ringFactory(Kamon.config)
+
+  /**
+    * The current index to the average ring buffer
+    */
+  private var averageIndex: Int = 0
+
+  /**
+    * Rebuilds the average ring, based on the current configuration
+    */
+  private def ringFactory(config: Config): Array[TickMetricSnapshot] = {
+    val tickInterval: Long = config.getDuration("kamon.metric.tick-interval").toMillis
+    val fullFrames: Int = (averagingWindow.toMillis.toFloat / tickInterval).ceil.toInt
+
+    // Warn for too low values
+    if (fullFrames < 3) {
```

Comment:
Let's make this magic number a bit more obvious :)

Reply:
I will add some comment above 😄